graph/src/main/scala/net/spals/dsa/graph/Graph.scala

```diff
@@ -17,10 +17,10 @@ object WeightedGraph {
 
 case class WeightedGraph[T, W](vertices: Set[Vertex[T]], edgeFunc: (Vertex[T], Vertex[T]) => Option[W]) {
 
-  def addEdge(v1: Vertex[T], v2: Vertex[T], w: W): WeightedGraph[T, W] = {
+  def addEdge(v1: Vertex[T], v2: Vertex[T], weight: W): WeightedGraph[T, W] = {
```

Comment:
Don't think there's a hard and fast rule. I thought `v1` and `v2` were ok because of their types.

Also, variables in case class constructors are automatically exposed through getters. And I thought `edge.weight` would make more sense to a caller than `edge.w`